internal/graph/reachableresources.go

```diff
@@ -360,21 +362,28 @@ func (crr *ConcurrentReachableResources) redispatchOrReport(
 		if foundResourceType.Namespace == parentRequest.ResourceRelation.Namespace &&
 			foundResourceType.Relation == parentRequest.ResourceRelation.Relation {
 			return parentStream.Publish(&v1.DispatchReachableResourcesResponse{
```

Comment:
optimized entrypoints will never return empty if there is at least one valid entrypoint in the unoptimized version. optimized entrypoints is different because it only follows a single child of the intersection, rather than all of them, because if we find a resource on one path of the intersection, it must be on all others for it to be valid